docs/tooling/profiling.md

```diff
@@ -0,0 +1,90 @@
+# Profiling on the Nao Robot
+
+We use the perf command to profile applications running on the Nao robot. It comes preinstalled in our robot image.
+
+!!!warning
+    Important: Ensure that your hulk binary includes debug symbols; without them, the profile will be unusable.
+    Enable debug symbols by adding the --profile with-debug option to either the pepsi upload or pepsi build command. This preserves compiler optimizations while retaining symbol information for profiling.
+
+### Step 1: SSH into the Robot
+
+To profile an application running on the Nao, you must SSH into the robot first:
+
+```bash
+pepsi shell <nao-number>
+```
+
+### Step 2: Record a Profile with perf
+
+To profile a running application (e.g., hulk), use:
+
+perf record --call-graph dwarf,8192 --aio -z --sample-cpu --mmap-pages 16M --pid $(pidof hulk) sleep 30
+
+This samples the stack traces of the hulk process for 30 seconds.
+
+!!!info
+    It's generally easiest to run perf as root, as otherwise various permissions and kernel knobs must be adjusted.
+
+This command will generate a perf.data file containing the recorded samples.
+
+### Step 3: Analyze the Profile with hotspot
+
+We use Hotspot to inspect the perf.data file.
+Rename the Binary
+
+Due to an automatic renaming during upload, your binary might be called hulk_nao. Hotspot expects it to match the process name (hulk), so rename it:
```

Comment:
we can actually also discuss, whether we just want to rename the binary via cargo? This might be possible?! -> issue for future iteration?